crates/pgsrv/src/handler.rs

```diff
@@ -233,20 +234,40 @@ impl ProtocolHandler {
             }
         }
 
-        let mut sess = match self
+        let mut vars = SessionVars::default();
+        // Set "system" variables first.
+        // Info
+        vars.user_id.set_and_log(user_id, VarSetter::System);
+        vars.user_name.set_and_log(user_name, VarSetter::System);
+        vars.connection_id.set_and_log(conn_id, VarSetter::System);
+        vars.database_id.set_and_log(db_id, VarSetter::System);
+        vars.database_name
+            .set_and_log(database_name, VarSetter::System);
+        // Limits
+        vars.max_datasource_count
+            .set_and_log(Some(max_datasource_count), VarSetter::System);
+        vars.memory_limit_bytes
+            .set_and_log(Some(memory_limit_bytes), VarSetter::System);
+        vars.max_tunnel_count
+            .set_and_log(Some(max_tunnel_count), VarSetter::System);
+        vars.max_credentials_count
+            .set_and_log(Some(max_credentials_count), VarSetter::System);
+
```

Comment:
Shouldn't we error (to the user) if we can't set any system var here? We don't want faulty sessions.